[DO NOT REVIEW] Add support for Azure Blob Storage and ADLS Gen2 in Hive connector (#25107)
prestodb/presto: presto-docs/src/main/sphinx/connector/hive.rst

pool*, increase the value of both ``hive.s3select-pushdown.max-connections`` and
the maximum connections configuration for the file system you are using.

Azure Configuration
-------------------

The Hive connector supports Azure Blob Storage (``wasb://``) and Azure Data Lake Storage Gen2 (``abfs://``).

> Configure Azure storage access in your catalog properties file (e.g., ``etc/catalog/hive.properties``).

Reviewer: ```suggestion
Configure Azure storage access in your catalog properties file, such as in ``etc/catalog/hive.properties``.
```
Avoid Latin abbreviations. See the [GitLab documentation style guide recommended word list entry for e.g.](https://docs.gitlab.com/development/documentation/styleguide/word_list/#eg) for alternatives. I've suggested one alternative here, please revise it if a different alternative fits better.

Azure Blob Storage
^^^^^^^^^^^^^^^^^^

.. code-block:: none

    hive.azure.wasb.storage-account=mystorageaccount
    hive.azure.wasb.access-key=<wasb-access-key>

ADLS Gen2 - Shared Key
^^^^^^^^^^^^^^^^^^^^^^

.. code-block:: none
    hive.azure.abfs.storage-account=mydatalake
    hive.azure.abfs.access-key=<abfs-access-key>

ADLS Gen2 - OAuth 2.0
^^^^^^^^^^^^^^^^^^^^^

. code-block:: none
    hive.azure.abfs.storage-account=mydatalake
    hive.azure.abfs.oauth.client-endpoint=https://login.microsoftonline.com/<tenant-id>/oauth2/token
    hive.azure.abfs.oauth.client-id=<oauth-client-id>
    hive.azure.abfs.oauth.client-secret=<your_client_secret_here>

Configuration Properties
^^^^^^^^^^^^^^^^^^^^^^^^

> ================================================= =====================================================================

Reviewer: The table in lines 737-757 isn't displaying in the local doc build. See screenshot. 

<img width="638" height="533" alt="Screenshot 2025-10-30 at 10 24 36 AM" src="https://github.com/user-attachments/assets/ef65244e-9523-4221-9e69-e2de660efed1" />

Property Name                                       Description
================================================= =====================================================================
``hive.azure.wasb.storage-account``                 Azure Blob Storage account name
                                                    (without ``.blob.core.windows.net suffix``)

``hive.azure.wasb.access-key``                      Azure Blob Storage access key

``hive.azure.abfs.storage-account``                 ADLS Gen2 storage account name

>                                                     (without ``.blob.core.windows.net suffix``)

Reviewer: `suffix` should not be within the formatting ticks

``hive.azure.abfs.access-key``                      Shared access key for ADLS Gen2

``hive.azure.abfs.oauth.client-endpoint``           OAuth 2.0 token endpoint(includes tenant ID)
                                                    (format: https://login.microsoftonline.com/<tenant-id>/oauth2/token

``hive.azure.abfs.oauth.client-id``                 OAuth 2.0 client/application ID

``hive.azure.abfs.oauth.client-secret``             OAuth 2.0 client secret
======================================================= ===============================================================

Configuration Requirement:
##########################

* WASB requires both ``wasb.storage-account`` and ``wasb.access-key`` to be set
* ABFS with shared key requires both ``abfs.storage-account`` and ``abfs.access-key`` to be set
* ABFS with OAuth requires ``abfs.storage-account``, ``abfs.oauth.client-endpoint``, ``abfs.oauth.client-id``, and
``abfs.oauth.client-secret`` (all four properties) to be set

Avoid hardcoding credentials in configuration files. Use environment variables or secret management systems.

URI Format
^^^^^^^^^^

Azure Blob Storage:

> .. code-block:: none

Reviewer: Please fix the formatting to remove the vertical gray line to the left of `wasb://<container>@<storage-account>.blob.core.windows.net/<path>` in line 774, and to the left of `abfs://<container>@<storage-account>.dfs.core.windows.net/<path>` in line 778. 

See screenshot. 
<img width="544" height="197" alt="Screenshot 2025-10-30 at 10 24 36 AM copy" src="https://github.com/user-attachments/assets/6cfe7360-0cce-45ff-9a50-5e85b40c86e7" />

pool*, increase the value of both ``hive.s3select-pushdown.max-connections`` and
the maximum connections configuration for the file system you are using.

Azure Configuration
-------------------

The Hive connector supports Azure Blob Storage (``wasbs://``) and Azure Data Lake Storage Gen2 (``abfss://``).
Configure Azure storage access in your catalog properties file, such as in ``etc/catalog/hive.properties``.

Azure Blob Storage
^^^^^^^^^^^^^^^^^^

.. code-block:: none

>     hive.azure.wasb.storage-account=mystorageaccount

Reviewer 2: The property names here use dots (e.g., `hive.azure.wasb.storage-account`) but the @Config annotations in HiveAzureConfig.java use hyphens (e.g., `hive.azure.wasb-storage-account`). These need to match or users will get configuration errors.

        totalprice double
    ) WITH (
        external_location = 'abfss://mycontainer@mydatalake.dfs.core.windows.net/orders/',
        format = 'PARQUET'
    );

Create Schema
#############

.. code-block:: none

    CREATE SCHEMA hive.azure_data
    WITH (location = 'abfss://mycontainer@mydatalake.dfs.core.windows.net/schemas/');

Query and Write
###############

.. code-block:: none

    -- Query existing data
    SELECT * FROM hive.default.orders;

    -- Write new data
    CREATE TABLE hive.azure_data.sales WITH (format = 'ORC')
    AS SELECT * FROM hive.default.orders WHERE orderdate > DATE '2024-01-01';

Troubleshooting
^^^^^^^^^^^^^^^

Authentication Errors
#####################

If you see ``403 Forbidden`` or ``401 Unauthorized``:

* Verify storage account names are correct (without ``.blob.core.windows.net`` or ``.dfs.core.windows.net`` suffix)
* Check if access keys or OAuth2 credentials are correct
* For OAuth2, ensure the service principal has Storage Blob Data Contributor or Reader role in Azure

Connection Errors
#################

If you see connection timeouts or ``UnknownHostException``:

* Verify network connectivity between Presto nodes and Azure

> * Check if Azure storage firewall rules allow access from Presto nodes

Reviewer 3: Will be good if we can provide some link here as well which explains how to check Azure firewall rules. Maybe a link from azure documentation itself should suffice.
